src/javascripts/ng-admin/es6/lib/DataStore/DataStore.js

```diff
@@ -0,0 +1,119 @@
+import Entry from "../Entry";
+
+class DataStore {
+    constructor() {
+        this._entries = {};
+    }
+
+    setEntries(name, entries) {
+        this._entries[name] = entries;
+
+        return this;
+    }
+
+    addEntry(name, entry) {
+        if (!(name in this._entries)) {
+            this._entries[name] = [];
+        }
+
+        this._entries[name].push(entry);
+    }
+
+    getEntries(name) {
+        return this._entries[name] || [];
+    }
+
+    getFirstEntry(name) {
+        return this._entries[name][0];
+    }
+
+    getChoices(field) {
+        var identifier = field.targetEntity().identifier().name();
+        var name = field.targetField().name();
+
+        return this.getEntries(field.targetEntity().uniqueId + '_choices').map(function(entry) {
+            return {
+                value: entry.values[identifier],
+                label: entry.values[name]
+            };
+        });
+    }
+
+    createEntry(entityName, identifier, fields) {
+        var entry = new Entry.mapFromRest(entityName, identifier, fields, {});
+
+        fields.forEach(function (field) {
```

Comment:
It's more like 3 times slower if you take a non-basic loop body. Using a `for` loop is a micro-optimization IMO, and makes the code less readable.